File: web3-libs/mod3/playground/aula3/get_transaction_logs.py
# 1. Instale e importe a biblioteca web3.py
from web3 import Web3
from pprint import pprint

# 2. Configure o provider
RPC_URL = "http://127.0.0.1:8545"
web3 = Web3(Web3.HTTPProvider(RPC_URL))

# 3. Obtenha o hash da transação via input
tx_hash = input("Digite o hash da transação: ")

try:
    transaction_receipt = web3.eth.get_transaction_receipt(tx_hash)
    print("Logs da Transação:")
    for log in transaction_receipt.logs:
        print("Log Address:", log.address)
        print("Log Topics:", log.topics)
        print("Log Data:", log.data)
        print("---")
except Exception as error:
    print("Erro ao buscar a transação:", error)

File: web3-libs/mod2/playground/aula3/getTransactionLogs.js
// 1. Instale e importe a biblioteca Web3.js
import Web3 from "web3";
import readline from "readline";

// 2. Configure o provider
const RPC_URL = "http://127.0.0.1:8545";
const web3 = new Web3(RPC_URL);

// 3. Obtenha o endereço da conta via readline input
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Digite o endereço da conta: ", (txHash) => {
  web3.eth
    .getTransactionReceipt(txHash)
    .then((transaction) => {
      console.log("Logs da Transação:", transaction.logs);
      transaction.logs.forEach((log) => {
        console.log("Log Address:", log.address);
        console.log("Log Topics:", log.topics);
        console.log("Log Data:", log.data);
      });
    })
    .catch((error) => {
      console.error("Erro ao buscar a transação:", error);
    });
  rl.close();
});
